src/index.ts

```diff
@@ -501,10 +511,17 @@ async function initComponents(): Promise<AppComponents> {
     chainId: collectionsChainId,
   })
 
+  const collectionsEthereumSales = createSalesComponent({
+    subgraph: collectionsEthereumSubgraph,
+    network: Network.MATIC, // we'use MATIC here so it uses the right fragment when querying
```

Comment:
i didn't follow why the network provided here is not ethereum?

Reply:
the fragment for the sale is later on picked by the `network` passed here. We want the MATIC fragment since the collections-ethereum-* graph has the same structure (specifically the `itemId` that I want to filter by). Does it make sense?

Reply:
ahh is it this? https://github.com/decentraland/nft-server/blob/master/src/ports/sales/utils.ts#L29-L42
so the `marketplace` subgraph has one schema, and the `collections-ethereum-*` and `collections-matic-*` have the other schema (with `searchItemId`)?

If so, maybe instead of relying on the `network` to determine whether to add the `searchItemId` or not, we should use a different property, because the assumtion on the network is wrong.

What do you think about adding a `useLegacySchema?: boolean` to the `createSalesComponent`, and let the 
`marketplaceSales` component pass that as `true` and the other sales components could just ignore the property. 

Then the `getSaleFragment` could replace the `network` param with a `useLegacySchema` too, and be like:

```ts
export const getSaleFragment = (useLegacySchema = false) => `
  fragment saleFragment on Sale {
    id
    type
    buyer
    seller
    price
    timestamp
    txHash
    ${useLegacySchema ? '' : 'searchItemId'}
    searchTokenId
    searchContractAddress
  }
`
```

what do you think?

Reply:
sounds good! Update to that :)

Reply:
ty 🙏